### ISSUE TITLE
Use stylelint to enfore css prefixes

### ISSUE THREAD

**Issue by anandthakker** [mapbox/mapbox-gl-js#4381]
To avoid regressions on fixes like https://github.com/mapbox/mapbox-gl-js/pull/4357, we should use https://github.com/stylelint/stylelint to enforce that every CSS class we define is prefixed with `mapbox`.

**Comment by asantos3026** [comment]
@anandthakker I'm looking in to this issue. Is there a preference on how stylelint should be used? Here are the options for stylelint. 

```
Decide how you want to use stylelint:
via the stylelint CLI
via a plugin for your build tool (gulp, webpack etc)
via a plugin for your text editor (atom, sublime text etc)
via the stylelint Node API
via the stylelint PostCSS plugin
```

**Comment by anandthakker** [comment]
@asantos3026 thanks for looking into it!

I think using the CLI is probably our best bet.  Using the stylelint CLI, we could add a `"lint-css"` script to [package.json](https://github.com/mapbox/mapbox-gl-js/blob/master/package.json#L116-L117), and then include that script in the [list of scripts run by `"test"`](https://github.com/mapbox/mapbox-gl-js/blob/master/package.json#L119)

**Comment by asantos3026** [comment]
I was just reading this blog and I came to the same conclusion 👍  

`Usage Link
Stylelint can be used in many ways: in the command line, in a build tool (such as Gulp, Grunt or Webpack), in a code editor or as a Git pre-commit hook for staged changes in Git repository. I will focus here on two ways.
COMMAND LINE LINK
Using the command line is useful when you want to lint a project that doesn’t have stylelint or you want to use stylelint in an npm script.
Install stylelint globally:
npm install stylelint -g
Then, it will be available everywhere in your terminal:
stylelint "styles/**/*.css"
This command will lint all CSS files in the styles directory and any of its subdirectories.
`

**Comment by asantos3026** [comment]
I'm new to developing and contributing to open source as part of my project this week :) I love Mapbox so wanted to contribute. Apologies for my novice question but for the style linter, is the declaration output in the css file supposed to be:

```
.mapbox-(class-name) {
  /* css code */
}
```

**Comment by anandthakker** [comment]
@asantos3026 questions to clarify an issue are very welcome!

Yes, I think what you've got there looks right to me: every CSS class in our output file should be in the form `.mapbox-(blah-blah-blah)`

**Comment by asantos3026** [comment]
Thanks @anandthakker. I added style lint and the naming convention seems to be working. How would I test?

**Comment by mollymerp** [comment]
closed by #4584
 thanks again @asantos3026 !
